README.md

```diff
@@ -107,49 +83,98 @@ Add required providers using the utility function `subformComponentProviders`:
 
-We also provide a utility function called `getControlsNames` which you can pass your `controls` to. This will let you define your `formControlName`s in the view in a safe way thanks to AoT. If you update your main interface, your form control but you forget about the view, you'll get an error.
+The NgxFormComponent base class automatically extracts your form control names and exposes them as a public member on `formControlNames`
 
-Then within the `.html`:
+Then within the `.html`, you can reference them like so:
 
 ```html
 <fieldset [formGroup]="formGroup" class="container">
-  <input type="text" placeholder="ID" [formControlName]="controlsNames.id" />
+  <input type="text" placeholder="ID" [formControlName]="formControlNames.id" />
 
-  <input type="number" placeholder="Price" [formControlName]="controlsNames.price" />
+  <input type="number" placeholder="Price" [formControlName]="formControlNames.price" />
 </fieldset>
 ```
+
+### Remapping Data
+
+It is a frequent pattern to have the data that you're trying to modify in a format that is incovenient to the angular
+forms structural constraints. For this reason, ngx-form-component offers a separate extended class `NgxSubFormRemapComponent`
+which will require you to define two interfaces - one to model the data going in to the form (which will be applied
+internally as `form.setValue()`), and the other to describe the interface of the value that will be set on the form.
+
+**You're always better off making your data structure better suit Angular forms, than abusing forms to fit your data pattern**
+
+For a complete example of this see `src/app/main/listing/vehicle-listing/vehicle-product.component.ts` (repeated below):
+
+```ts
+interface OneVehicleForm {
+  speeder: Speeder;
+  spaceship: Spaceship;
+  vehicleType: VehicleType;
+}
+
+@Component({
+  selector: 'app-vehicle-product',
+  templateUrl: './vehicle-product.component.html',
+  styleUrls: ['./vehicle-product.component.scss'],
+  providers: subformComponentProviders(VehicleProductComponent),
+})
+export class VehicleProductComponent extends NgxSubFormRemapComponent<OneVehicle, OneVehicleForm> {
+  protected formControls: Controls<OneVehicleForm> = {
+    speeder: new FormControl(null),
+    spaceship: new FormControl(null),
+    vehicleType: new FormControl(null, { validators: [Validators.required] }),
+  };
+
+  public VehicleType = VehicleType;
+
+  protected transformToFormGroup(obj: OneVehicle): OneVehicleForm {
+    return {
+      speeder: obj.vehicleType === VehicleType.SPEEDER ? obj : null,
+      spaceship: obj.vehicleType === VehicleType.SPACESHIP ? obj : null,
+      vehicleType: obj.vehicleType,
+    };
+  }
+
+  protected transformFromFormGroup(formValue: OneVehicleForm): OneVehicle {
+    switch (formValue.vehicleType) {
+      case VehicleType.SPEEDER:
+        return formValue.speeder;
+      case VehicleType.SPACESHIP:
+        return formValue.spaceship;
+    }
```

Comment:
Might be cool having an `UnreachableCase`

Reply:
Maybe, I kinda feel like demos should not introduce new concepts to people. Either way, not a dealbreaker